uaclient/system.py

```diff
@@ -518,3 +518,34 @@ def ensure_folder_absent(folder_path: str) -> None:
     if os.path.exists(folder_path):
         logging.debug("Removing folder: %s", folder_path)
         rmtree(folder_path)
+
+
+def is_dgx_os() -> bool:
+    return os.path.exists("/etc/dgx-release")
+
+
+def is_dgx_smbios() -> bool:
+    try:
+        baseboard_man_out, _ = subp(
+            ["dmidecode", "-s", "baseboard-manufacturer"]
```

Comment:
We probably need to add a runtime dependency on the `dmidecode` package.

Reply:
We need to keep dependencies small as this is also in ubuntu-minimal and many things - like dmidecode are not.

It is in "ubuntu-standard" so we can assume most Ubuntu and derivatives have it.
A better fallback would be "if no dmidecode is around, check sysfs".

We lack a DGX system atm, @dannf could you check if there is anything reasonable for that in `/sys/class/dmi/id/*` ?

Reply:
Good call @cpaelzer  - looks like the same values are available in `/sys/class/dmi/id/board_name` and `/sys/class/dmi/id/board_vendor`:

```
ubuntu@exotic-skunk:~$ cat /sys/class/dmi/id/board_name
DGX-1 with V100
ubuntu@exotic-skunk:~$ cat /sys/class/dmi/id/board_vendor
NVIDIA
```

Verified with bionic on a DGX-1 and jammy on a DGX A100 (oldest system/oldest supported kernel + newest system/newest supported kernel). So +1 on using that instead of `dmidecode`.